program fgeti_test;
#include( "stdlib.hhf" )

var
	b			:blob.blob;
	inputF		:real80;
	
begin fgeti_test;

	blob.alloc( 8192 );
	mov( eax, b );

	// Testing blob.geti32
	
	try
	
		blob.load( "bgeti32.data", b );

		blob.geti32( b );
		if( eax <> 1 ) then
		
			raise(1011);
		
		endif;	
		
		blob.geti32( b );
		if( eax <> -2 ) then
		
			raise(1012);
		
		endif;	
		
		blob.geti32( b );
		if( eax <> -2147483648 ) then
		
			raise(1013);
		
		endif;	
		
		blob.geti32( b );
		if( eax <> 2147483647 ) then
		
			raise(1014);
		
		endif;


		
		try
		
			blob.geti32( b );

		  unprotected
		  
		  	// We we got to this point, we failed to raise a conversion error.
			
		  	Raise( 1005 );
			
		  exception( ex.ConversionError )
		  
		  	// Okay, we succeeded if we got this exception.
			
		  anyexception
		  
		  	// Anything else? Pass it on.
			
		  	Raise( 1006 );
			
		endtry;
		blob.readLn( b );	// Clean up after ourselves.
		
		
		try
		
			blob.geti32( b );

		  unprotected
		  
		  	// We we got to this point, we failed to raise an overflow error.
			
		  	Raise( 1015 );
			
		  exception( ex.ValueOutOfRange )
		  
		  	// Okay, we succeeded if we got this exception.
			
		  anyexception
		  
		  	// Anything else? Pass it on.
			
		  	Raise( 1016 );
			
		endtry;
		blob.readLn( b );
		
		
		// Better be at end of file.
		
		if( !blob.eof( b )) then
		
			raise(1010);
			
		endif;
		stderr.put( "blob.geti32 succeeded!" nl );
		
	  anyexception
	  
		stderr.put
		( 
			nl nl 
			"***************************************************" nl
			"blob.geti32 failed! eax=", (type uns32 eax), nl
			"Data read did not match data expected in file" nl
			"***************************************************" nl
			nl 
		);
		stdout.put( "Press Enter to Continue: " );
		stdin.readLn();
		os.exitProcess(1);
		
	endtry;








	// Testing blob.geti16
	
	try
	
		blob.load( "bgeti16.data", b );

		blob.geti16( b );
		if( ax <> 0 ) then
		
			raise(1021);
		
		endif;	
		
		blob.geti16( b );
		if( ax <> 1 ) then
		
			raise(1022);
		
		endif;	
		
		blob.geti16( b );
		if( ax <> -1 ) then
		
			raise(1023);
		
		endif;	
		
		blob.geti16( b );
		if( ax <> -32768 ) then
		
			raise(1024);
		
		endif;	
		
		blob.geti16( b );
		if( ax <> 32767 ) then
		
			raise(1025);
		
		endif;
		
		try
		
			blob.geti16( b );

		  unprotected
		  
		  	// We we got to this point, we failed to raise an overflow error.
			
		  	Raise( 1026 );
			
		  exception( ex.ValueOutOfRange )
		  
		  	// Okay, we succeeded if we got this exception.
			
		  anyexception
		  
		  	// Anything else? Pass it on.
			
		  	Raise( 1027 );
			
		endtry;
		blob.readLn( b );
		
		
		// Better be at end of file.
		
		if( !blob.eof( b )) then
		
			raise(1020);
			
		endif;
		stderr.put( "blob.geti16 succeeded!" nl );
		
	  anyexception
	  
		stderr.put
		( 
			nl nl 
			"***************************************************" nl
			"blob.geti16 failed! eax=", (type uns32 eax), nl
			"Data read did not match data expected in file" nl
			"***************************************************" nl
			nl 
		);
		stdout.put( "Press Enter to Continue: " );
		stdin.readLn();
		os.exitProcess(1);
		
	endtry;





	// Testing blob.geti8
	
	try
	
		blob.load( "bgeti8.data", b );

		blob.geti8( b );
		if( al <> 0 ) then
		
			raise(1031);
		
		endif;	
		
		blob.geti8( b );
		if( al <> 1 ) then
		
			raise(1032);
		
		endif;	
		
		blob.geti8( b );
		if( al <> -1 ) then
		
			raise(1033);
		
		endif;	
		
		blob.geti8( b );
		if( (type int8 al) <> -128 ) then
		
			raise(1034);
		
		endif;
		
		blob.geti8( b );
		if( al <> 127 ) then
		
			raise(1035);
		
		endif;
		
		try
		
			blob.geti8( b );

		  unprotected
		  
		  	// We we got to this point, we failed to raise an overflow error.
			
		  	Raise( 1036 );
			
		  exception( ex.ValueOutOfRange )
		  
		  	// Okay, we succeeded if we got this exception.
			
		  anyexception
		  
		  	// Anything else? Pass it on.
			
		  	Raise( 1037 );
			
		endtry;
		blob.readLn( b );
		
		
		// Better be at end of file.
		
		if( !blob.eof( b )) then
		
			raise(1030);
			
		endif;
		stderr.put( "blob.geti8 succeeded!" nl );
		
	  anyexception
	  
		stderr.put
		( 
			nl nl 
			"***************************************************" nl
			"blob.geti8 failed! eax=", (type uns32 eax), nl
			"Data read did not match data expected in file" nl
			"***************************************************" nl
			nl 
		);
		stdout.put( "Press Enter to Continue: " );
		stdin.readLn();
		os.exitProcess(1);
		
	endtry;




			
end fgeti_test;

													